MessageJsonAdapter added to support Gson polymorphism conversion.

teamagam1 opened the pull request:
Few uncertainties:
1. protected members initialize with 'p' instead of 'm'?
2. each MessageType has it's own getter and setter. even though they all use the same protected mContent. do we want to change that? 
3. delete MessageTypeMarkerInterface? how do we mark the content?

teamagam1 committed:
MessageJsonAdapter added to support Gson polymorphism conversion.

mContent is now a protected member of Message.

each MessageType has its own getter and setter so it is will be clear what you get.

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/Message.java:
```diff
@@ -1,80 +1,69 @@
-public class Message<T extends MessageContentInterface>{
+public abstract class Message<T>{
+
+    @StringDef({TEXT, LAT_LONG, USER_LOCATION})
```
You should add the @Retention annotation to explicitly define the retention policy.

IMO its a SOURCE retention policy (at compile time - to avoid performance issues)

BarAgam replied:
!fix

teamagam1 replied:
!fixed

BarAgam commented:
protected members initialize with 'p' instead of 'm'?

Nah, m is prefixes all members but statics

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageJsonAdapter.java:
```diff
@@ -0,0 +1,65 @@
+import java.util.TreeMap;
+
+/**
+ * Created by Yoni on 4/19/2016.
```
class documentation !fix

teamagam1 replied:
!fixed

BarAgam commented:
each MessageType has it's own getter and setter. even though they all use the same protected mContent. do we want to change that?

I think we do. getContent is OK.
An alternative is to implement set/getContent in the abstract class, and each implementation adds another getter/setter with specific naming

BarAgam commented:
delete MessageTypeMarkerInterface? how do we mark the content?

delete

teamagam1 committed:
added getContent to Message class replaces all specific getters..
added RetentionPolicy.SOURCE for MessageType.

removed MessageTypeMarkerInterface

teamagam1 committed:
Merge remote-tracking branch 'origin/feature/gps_and_location_sharing' into task/gps_and_location_manipulation/new_message_class

# Conflicts:
#	app/src/test/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageUserLocationUnitTest.java
